hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -310,10 +302,11 @@ def _define_setting(
             options = tuple(options)
             assert default in options
 
-        if future_default is not_set:
+        # No branch on these conditions because we don't have any deprecated
```

Comment:
It might be easier to write a new test for this with a fake setting than to keep track of when to add and remove the `pragma: no branch`.

If you want it, I think this works:

```
from hypothesis import _settings
def test_deprecated_setting():
    fake_all_settings = {}
    with mock._patch_object(_settings, 'all_settings', fake_all_settings):
        with mock._patch_object(settings, '_settings__definitions_are_locked', False):
            settings._define_setting(
                "fake_deprecated_setting",
                default=60,
                description="fake deprecated setting",
                deprecation_message="is deprecated!",
                deprecated_since="2017-11-02",
                future_default=unlimited,
                hide_repr=False)
        assert fake_all_settings['fake_deprecated_setting'].future_default == unlimited
        assert fake_all_settings['fake_deprecated_setting'].hide_repr == False

```

Reply:
Looking into this in more detail, we can actually delete these entirely.  Our current practice for deprecations is to implement the new behaviour immediately, and only keep the deprecated interface as a warning-emitting shim.

Under this model, `future_default` is *always*` `not_set`, and we will never have a `timeout`-like case where we need to show the default value of a deprecated setting because it still has an effect.

Reply:
Cool!

Reply:
Cool!